product_harmonized_system_heading/models/hs_code.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2023 ForgeFlow S.L. (https://www.forgeflow.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools import config
+
+
+class HSCode(models.Model):
+    _inherit = "hs.code"
+
+    hs_code_heading_id = fields.Many2one(comodel_name="hs.code.heading", readonly=True)
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        if not config["test_enable"] or self.env.context.get("check_hs_code_heading"):
```

Comment:
Instead of this, just do `hs_code_heading_id` a computed writable field, but removing the error if not found.

Reply:
I've just changed it now to work in the following way:
- When an H.S. Code is created, it will try to find the corresponding Heading, if none is found, it will automatically create a new one, instead of triggering an error to then manually create the Heading.
- If we edit the Local Code, we also call the `_get_hs_code_heading_id` method to get the new Heading.

Doing it like this, we don't need to adapt any tests and by optionally having this module, the user won't need to go through any additional steps.